packages/site/cms-site/src/sitePreview/SitePreviewProvider.tsx

```diff
@@ -1,21 +1,19 @@
-import { useRouter } from "next/router";
+"use client";
+import { usePathname } from "next/navigation";
 import * as React from "react";
 
 import { PreviewContext } from "../preview/PreviewContext";
 import { sendSitePreviewIFrameMessage } from "./iframebridge/sendSitePreviewIFrameMessage";
 import { SitePreviewIFrameLocationMessage, SitePreviewIFrameMessageType } from "./iframebridge/SitePreviewIFrameMessage";
 
 const SitePreview: React.FunctionComponent<{ children: React.ReactNode }> = ({ children }) => {
-    const router = useRouter();
+    const pathname = usePathname();
 
     React.useEffect(() => {
         function sendUpstreamMessage() {
-            const url = new URL(router.asPath, window.location.origin);
```

Comment:
useRouter.asPath did include search params, usePathname doesn't. And next h as no api to get search, that's why I added location.search for now.

This doesn't update if search changes!

Reply:
How about [useSearchParams](https://nextjs.org/docs/app/api-reference/functions/use-search-params)?

Reply:
ah, yes of course :) sorry